controllers/dspipeline_fake_controller.go

```diff
@@ -0,0 +1,90 @@
+//go:build test_all || test_unit
+// +build test_all test_unit
+
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	dspav1alpha1 "github.com/opendatahub-io/data-science-pipelines-operator/api/v1alpha1"
+	buildv1 "github.com/openshift/api/build/v1"
+	imagev1 "github.com/openshift/api/image/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+)
+
+func NewFakeController() *DSPAReconciler {
+	// Setup Fake Client Builder
+	fake_builder := fake.NewClientBuilder()
+
+	// Create Scheme
+	fake_scheme := scheme.Scheme
+	utilruntime.Must(clientgoscheme.AddToScheme(fake_scheme))
+	utilruntime.Must(buildv1.AddToScheme(fake_scheme))
+	utilruntime.Must(imagev1.AddToScheme(fake_scheme))
+	utilruntime.Must(routev1.AddToScheme(fake_scheme))
+	utilruntime.Must(dspav1alpha1.AddToScheme(fake_scheme))
+	fake_builder.WithScheme(fake_scheme)
+
+	// Build Fake Client
+	fake_client := fake_builder.Build()
+
+	// Generate DSPAReconciler using Fake Client
+	r := &DSPAReconciler{
+		Client:        fake_client,
+		Log:           ctrl.Log.WithName("controllers").WithName("ds-pipelines-controller"),
+		Scheme:        fake_scheme,
+		TemplatesPath: "../config/internal/",
+	}
+
+	return r
+}
+
+func CreateNewTestObjects() (context.Context, *DSPAParams, *DSPAReconciler) {
+	return context.Background(), &DSPAParams{}, NewFakeController()
+}
+
+func (r *DSPAReconciler) IsResourceCreated(ctx context.Context, obj client.Object, name, namespace string) (bool, error) {
+	// Fake Request for verification
+	nn := types.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}
+
+	// Fetch
+	err := r.Get(ctx, nn, obj)
+
+	// Err shouldnt be thrown if resource exists
+	// TODO: implement better verification
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			return false, nil
+		} else {
+			return false, err
+		}
+	}
+	return true, nil
+}
```

Comment:
this is extending the reconciler object right? should this be in a mock controller file?

Reply:
...this is that file??

Reply:
I meant that this is a "non test" object being extended in a "test file", it will be accessible in non test code, so it should probably be extended in a file that is not *_test.go, or for a fake controller, but I don't feel too strongly about it, we  can keep it here.